src/implicit_function.jl

```diff
@@ -15,21 +33,62 @@ This amounts to solving a linear system `A * J = -B`, where `A ∈ ℝᵈˣᵈ`,
 - `conditions::C`: callable of the form `(x,y,z) -> F(x,y,z)`
 - `linear_solver::L`: callable of the form `(A,b) -> u` such that `Au = b`, must be taken from Krylov.jl
 """
-struct ImplicitFunction{F,C,L}
+struct ImplicitFunction{F<:Forward,C<:Conditions,L}
```

Comment:
Should we check that both `Forward` and `Conditions` have the same byproduct behavior?
```julia
struct ImplicitFunction{BP,F<:Forward{BP},C<:Conditions{BP},L}
```

Reply:
yes I think that makes sense